### PR TITLE
Auto-generating Django migrations using Django's own machinery

### PR DESCRIPTION
**Pull request by ewjoachim**
Up for discussion! I think I prefer the idea of having Django more autonomous, but:
- Django's code is not very modular, so I can have minimal code
- This means the corresponding code is not very readable
- And I have to reimplement a few details that are already implemented in Django's own code but I couldn't reuse.
- That being said, the code here is more testable than the script it replaces (well... differently testable. The script was not easily testable because it was not part of the python package. That version is more easily testable in this way, but we'll need more integration tests, vs unit tests)

It's kind of funny because Django here does a lot of very smart things, but they have so many unrelated code in the same functions, that there's always one little detail that makes many things not reusable. That being said, this code in django was most probably not supposed to be hacked, so I expected it to be hard.

Hmm. I'm wondering about grouping all the missing SQL migrations in a single django migrationwith multiple operations. This would simplify the code a lot !

### Successful PR Checklist:

- [ ] Tests
  - [ ] (not applicable?)
- [x] Documentation
  - [ ] (not applicable?)
- [ ] Had a good time contributing? < I've known better
- [ ] (Maintainers: add PR labels)

### CODE REVIEW

File: procrastinate/contrib/django/management/commands/makemigrations.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+def version_from_string(version_str) -> Tuple:
+    return tuple(int(e) for e in version_str.split("."))
+
+
+@attr.dataclass(frozen=True)
+class ProcrastinateMigration:
+    filename: str
+    name: str
+    version: Tuple
+    body: str
```
Comment (ewjoachim) [inline procrastinate/contrib/django/management/commands/makemigrations.py:82]: index: int

File: procrastinate/contrib/django/management/commands/makemigrations.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+def get_all_migrations() -> Iterable[ProcrastinateMigration]:
+    all_files = utils.list_migrations()
+    migrations = [
+        ProcrastinateMigration.from_string(path=e)
+        for e in all_files
+        if e.suffix == ".sql"
+    ]
+
+    return sorted(migrations, key=lambda x: x.version)
```
Comment (ewjoachim) [inline procrastinate/contrib/django/management/commands/makemigrations.py:107]:
```suggestion
    return sorted(migrations, key=lambda x: (x.version, x.index))
```

File: procrastinate/contrib/django/management/commands/makemigrations.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+def get_existing_migrations(
+    loader: MigrationLoader, all_migrations: Iterable[ProcrastinateMigration]
+) -> Iterable[ProcrastinateMigration]:
+    migration_from_body = {m.body: m for m in all_migrations}
+    for (app, name), migration in loader.disk_migrations.items():
+        if app != "procrastinate":
+            continue
+        for operation in migration.operations:
+            if not isinstance(operation, migrations.RunSQL):
```
Comment (ewjoachim) [inline procrastinate/contrib/django/management/commands/makemigrations.py:118]:
```suggestion
            if not isinstance(operation, migrations.RunProcrastinateFile):
```

**Comment by ewjoachim** [comment]
@elemoine Ready for review !

File: docs/howto/django.rst [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+Install procrastinate with:
+
+.. code-block:: console
+
+    $ (venv) pip install 'procrastinate[django]'
+
+This tells pip to consider the extra dependencies from the group of dependencies named
+`django`. For now, this group only contains Django itself, which you likely already
+have in your project's dependencies. So why bother ?
```
Comment (elemoine) [inline docs/howto/django.rst:12]:
```suggestion
have in your project's dependencies. So why bother?
```

File: docs/howto/django.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+Specifying your dependency to the "`django` extras" will ensure that your Django version
```
Comment (elemoine) [inline docs/howto/django.rst:14]: Do you really want to use single back-ticks around `django` here?

File: docs/howto/django.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+Configure your procrastinate app from your Django settings::
+
+    import procrastinate
+    from procrastinate.contrib.django import connector_params
+
+    app = procrastinate.App(
+        connector=procrastinate.Psycopg2Connector(**connector_params())
+    )
+
+(See `connector` for more on how to instanciate your connector)
```
Comment (elemoine) [inline docs/howto/django.rst:37]:
```suggestion
(See `connector` for more on how to instanciate your connector.)
```

File: docs/howto/django.rst [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+Procrastinate comes with its own migrations so don't forget to call ``./manage.py
```
Comment (elemoine) [inline docs/howto/django.rst:39]: s/call/run ?

**Review by elemoine — APPROVED** [review]
A few minor comments, but it looks good to me.

File: docs/howto/django.rst (hunk 5 above)
Comment (ewjoachim) [reply docs/howto/django.rst:14]: Ah I meant double backticks.

**Comment by elemoine** [comment]
@ewjoachim you may want to add PR labels.

**Pull request merged**